src/util/Cache.h

```diff
@@ -227,6 +226,37 @@ class FlexibleCache {
     // TODO<joka921>:: implement this functionality
   }
 
+  MemorySize getMaxSizeSingleEntry() const noexcept {
+    return _maxSizeSingleEntry;
+  }
+
+  void recomputeSize(const Key& key) {
+    // Pinned entries must not be dynamic in nature
+    AD_CONTRACT_CHECK(!containsPinned(key));
+    if (!containsNonPinned(key)) {
+      return;
+    }
+    auto newSize = _valueSizeGetter(*(*this)[key]);
+    auto& oldSize = _sizeMap.at(key);
+    // Entry has grown too big to completely keep within the cache or we can't
+    // fit it in the cache
+    if (_maxSizeSingleEntry < newSize ||
+        _maxSize - std::min(_totalSizePinned, _maxSize) < newSize) {
+      erase(key);
+      return;
+    }
+
+    if (newSize >= oldSize) {
+      _totalSizeNonPinned += newSize - oldSize;
+    } else {
+      _totalSizeNonPinned -= oldSize - newSize;
+    }
+    oldSize = newSize;
```

Comment:
seems like dead code.

Reply:
Oh, it is the mutable ref into the size map, then please rename it to
`sizeInSizeMap`or something, else this looks weird.